src/options.cpp

```diff
@@ -256,10 +257,27 @@ void long_usage(char const *arg0, bool verbose)
 
 } // anonymous namespace
 
+static bool compare_prefix(std::string const &str,
+                           std::string const &prefix) noexcept
+{
+    return std::strncmp(str.c_str(), prefix.c_str(), prefix.size()) == 0;
+}
+
 std::string database_options_t::conninfo() const
 {
+    if (compare_prefix(db, "postgresql://") ||
+        compare_prefix(db, "postgres://")) {
+        return db;
+    }
+
     std::string out{"fallback_application_name='osm2pgsql'"};
 
+    if (std::strchr(db.c_str(), '=') != nullptr) {
```

Comment:
Would this cause problems with properly escaped DB names containing a =?

Reply:
Yes, it probably would. But how likely is that? I haven't looked at the code in `psql`, but I implemented this from their man page and that says: `If this parameter contains an = sign...`